### PR TITLE
Documentation overhaul

### PR DESCRIPTION
**Pull request by danielballan**
This PR...

* Overhauls the style, using Twitter's [bootstrap](http://getbootstrap.com/) CSS/JS framework for fluid (mobile-friendly) layouts and cross-browser support for interactive elements.
* Adds introductory notes outlining the project's purpose and its design philosophy.
* Adds installation instructions. *These are, of course, provisional. They use the tacaswell binstar channel. If including that in the docs, even just for now, is problematic, it can be removed.*
* Adds a framework for generating a tutorial with examples from IPython notebooks.
* Fixes some simple but important bugs discovered when building the docs. See commit messages.

This PR currently contains no examples. I propose merging this framework and then processing examples in separate PRs.

No content has been removed, and I think this is an improvement on the existing docs, even at this stage. Let me know if it is OK to force-push the built docs to the gh-pages branch.

### CODE REVIEW

File: doc/Makefile [hunk 1]
```diff
@@ -85,17 +88,17 @@ qthelp:
 	@echo
 	@echo "Build finished; now you can run "qcollectiongenerator" with the" \
 	      ".qhcp project file in $(BUILDDIR)/qthelp, like this:"
-	@echo "# qcollectiongenerator $(BUILDDIR)/qthelp/NSLS2.qhcp"
+	@echo "# qcollectiongenerator $(BUILDDIR)/qthelp/trackpy.qhcp"
```
Comment (ericdill) [inline doc/Makefile]: `trackpy` -> `scikit-xray`

File: doc/conf.py [hunk 2]
```diff
@@ -14,32 +13,39 @@
...
+              'sphinx.ext.mathjax',
+              'sphinx.ext.autosummary',
+              'sphinx.ext.extlinks',
+              'plot_generator',
+              'numpydoc',
+              'ipython_directive',
+              'ipython_console_highlighting',
+             ]
+
+extlinks = {'issue': ('https://github.com/soft-matter/scikit-xray/issues/%s',
```
Comment (ericdill) [inline doc/conf.py]: `soft-matter` -> `Nikea`

File: doc/index.rst [hunk 3]
```diff
@@ -1,22 +1,92 @@
...
+(Think LabView for X-ray image processing and data analysis.)
+
+
+The :ref:`tutorial <tutorial>` walks through several experiments from start to
+finish -- from the raw data to a publication-style plot. You can also browse
+the :ref:`API reference <api_ref>` for a quick overview of all the available
+tools.
+
+To check out the code, report a bug, or contribute a new feature, please visit
+the `github repository <https://github.com/soft-matter/trackpy>`_.
```
Comment (ericdill) [inline doc/index.rst]: `https://github.com/soft-matter/trackpy` -> `https://github.com/Nikea/scikit-xray`

File: doc/installation.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+    ipython notebook
+
+This will automatically open a browser tab, ready to interpret Python code.
+To get started, check out the links to tutorials at the top of this document.
+
+More Information for Experienced Python Users
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We strongly recommend using conda install trackpy, as described above,
```
Comment (ericdill) [inline doc/installation.rst]: `trackpy` -> `scikit-xray`

**Comment by danielballan** [comment]
grep confirms that there are no more references to "trackpy" or "soft-matter" lingering in there.

**Comment by ericdill** [comment]
Looks good, thanks Dan!

**Pull request merged**